### ISSUE TITLE
Copy to clipboard for How to Cite

### ISSUE THREAD

**Issue by StephDriver** [BirkbeckCTP/janeway#4236]
How to cite options are displayed with an expectation that the user will likely want to use the text elsewhere - i.e will want to copy it.

There should be a 'copy to clipboard' button to make this process more accessible.  Currently copying relies on the user to be able to select the appropriate text.  This can quite time-consuming and awkward for those relying on assistive technologies, where one might have to navigate to the start and end, checking each is correct. 

It is commonplace for there to be 'copy to clipboard' buttons on websites for content expected to be copied to address this.

**Comment by joemull** [comment]
In discussion we decided we do want this copy button, and we think the text sent to the clipboard should contain the full URL of the DOI with an anchor tag.

[Crossref](https://www.crossref.org/blog/accessibility-for-crossref-doi-links-call-for-comments-on-proposed-new-guidelines/) provides some guidelines on this that suggest forming the link like this:

```html
<a href="https://doi.org/10.5555/12345678" aria-label="DOI for Toward a Unified Theory of High-Energy Metaphysics: Silly String Theory">https://doi.org/10.5555/12345678</a>
```

**Comment by StephDriver** [comment]
> In discussion we decided we do want this copy button, and we think the text sent to the clipboard should contain the full URL of the DOI with an anchor tag.
> 
> [Crossref](https://www.crossref.org/blog/accessibility-for-crossref-doi-links-call-for-comments-on-proposed-new-guidelines/) provides some guidelines on this that suggest forming the link like this:
> 
> <a href="https://doi.org/10.5555/12345678" aria-label="DOI for Toward a Unified Theory of High-Energy Metaphysics: Silly String Theory">https://doi.org/10.5555/12345678</a>

Note, as part of this update, I have also standardised our syntax for DOI by creating a new `doi_display.html` template which takes in the document title and either the identifier, or the full url, and then formats the link including the aria-label.
